File: docs/pet.js
// Pet.js
export class Pet {
    constructor() {
        this._petType = 'none';
        this._petName = 'none';
        this._petAge = 0;
        this._dogSpaces = 30;
        this._catSpaces = 12;
        this._daysStay = 0;
        this._amountDue = 0.0;
    }

    // Getters
    get petType() { return this._petType; }
    get petName() { return this._petName; }
    get petAge() { return this._petAge; }
    get dogSpaces() { return this._dogSpaces; }
    get catSpaces() { return this._catSpaces; }
    get daysStay() { return this._daysStay; }
    get amountDue() { return this._amountDue; }

    // Setters
    set petType(type) { this._petType = type; }
    set petName(name) { this._petName = name; }
    set petAge(age) { this._petAge = age; }
    set dogSpaces(n) { this._dogSpaces = n; }
    set catSpaces(n) { this._catSpaces = n; }
    set daysStay(days) { this._daysStay = days; }
    set amountDue(amt) { this._amountDue = amt; }

    // Render to the console
    print() {
        console.log(`Pet type: ${this.petType}`);
        console.log(`Pet name: ${this.petName}`);
        console.log(`Pet age: ${this.petAge}`);

        if (this.petType.toLowerCase() === 'dog') {
            console.log(`Dog spaces: ${this.dogSpaces}`);
        } else if (this.petType.toLowerCase() === 'cat') {
            console.log(`Cat spaces: ${this.catSpaces}`);
        }

        console.log(`Days stay: ${this.daysStay}`);
        console.log(`Amount due: ${this.amountDue}`);
    }
}

File: Original Code/Pet.java

public class Pet {

	private String petType;
	private String petName;
	private int petAge;
	private int dogSpaces;
	private int catSpaces;
	private int daysStay;
	private double amountDue;
		
	/*Constructor*/
	public Pet() {
		petType = "none";
		petName = "none";
		petAge = 0;
		dogSpaces = 30;
		catSpaces = 12;
		daysStay = 0;
		amountDue = 0.0;
	}
	
	/*Setters*/
	public void setPetType(String type) {
		petType = type;
	}
	
	public void setPetName(String name) {
		petName = name;
	}
	
	public void setPetAge(int age) {
		petAge = age;
	}
	
	public void setDogSpaces (int dogSpace) {
		dogSpaces = dogSpace;
	}
	
	public void setCatSpaces (int catSpace) {
		catSpaces = catSpace;
	}
	
	public void setDaysStay (int days) {
		daysStay = days;
	}
	
	public void setAmountDue (double amount) {
		amountDue = amount;
	}
	
	/*Getters*/
	public String getPetType() {
		return petType;
	}
	
	public String getPetName() {
		return petName;
	}
	
	public int getPetAge() {
		return petAge;
	}
	
	public int getDogSpaces() {
		return dogSpaces;
	}
	
	public int getCatSpaces() {
		return catSpaces;
	}
	
	public int getDaysStay() {
		return daysStay;
	}
	
	public double getAmountDue() {
		return amountDue;
	}
	
	public void print() {
		System.out.println("Pet type: " + petType);
		System.out.println("Pet name: " + petName);
		System.out.println("Pet age: " + petAge);
		if(petType == "dog" || petType == "Dog") {
			System.out.println("Dog spaces: " + dogSpaces);
		if(petType == "cat" || petType == "Cat") {
			System.out.println("Cat spaces: " + catSpaces);
		}
		System.out.println("Days stay: " + daysStay);
		}
		System.out.println("Amount due: " + amountDue);
	}

}
